LaTeX/LaTeX.sublime-syntax

```diff
@@ -198,6 +208,153 @@ contexts:
+    - match: '[movsb]'
+      scope: storage.type.latex
+    - match: t
+      scope: storage.type.latex
+      push: xparse-args-token
+
+    # delimited arguments without default
+    - match: '[rd]'
+      scope: storage.type.latex
+      push:
+        - xparse-args-token
+        - xparse-args-token
+
+    # delimited arguments with default
+    - match: '[RD]'
+      scope: storage.type.latex
+      push:
+        - xparse-optarg-value
+        - xparse-args-token
+        - xparse-args-token
+
+    # optional argument with default
+    - match: O
+      scope: storage.type.latex
+      push: xparse-optarg-value
+
+    # embellishment argument
+    - match: e
+      scope: storage.type.latex
+      push: xparse-embellishment
+
+    # embellishment argument with default
+    - match: E
+      scope: storage.type.latex
+      push:
+        - xparse-optarg-value
+        - xparse-embellishment
+
+  xparse-embellishment:
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.latex
+      set: xparse-embellishment-content
+    - match: (?=\S)
+      set: xparse-args-token
+    - include: paragraph-pop
+
+  xparse-embellishment-content:
+    - match: (?!\})
+      push: xparse-args-token
+    - include: brace-group-end
+    - include: paragraph-pop
+    - include: else-pop
+
+  xparse-args-token:
+    - meta_include_prototype: false
+    - include: unmatched-brace-pop
+    - match: (?={)
+      set:
+        - xparse-args-wrapped-token
+        - brace-group-begin
+    - match: (?=\S)
+      set: xparse-args-bare-token
+    - include: paragraph-pop
+
+  xparse-args-bare-token:
+    # either a command
+    - match: (\\){{cmdname}}{{endcs}}
+      scope: constant.other.token.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      pop: 1
+    # or a single character
+    - match: \S
+      scope: constant.character.latex constant.other.token.latex
+      pop: 1
+    - include: paragraph-pop
+    - include: else-pop
+
+  xparse-args-wrapped-token:
+    - meta_scope: meta.group.brace.latex
+    - include: brace-group-end
+    # either a command
+    - match: (\\){{cmdname}}{{endcs}}
+      scope: constant.other.token.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+    # or a single character
+    - match: \S
+      scope: constant.character.latex constant.other.token.latex
+    - include: paragraph-pop
+
+  xparse-optarg-value:
+    - meta_include_prototype: false
+    - include: unmatched-brace-pop
+    - match: (?={)
+      set:
+        - xparse-optarg-wrapped-value
+        - brace-group-begin
+    - match: (?=\S)
+      set: xparse-optarg-bare-value
+    - include: paragraph-pop
+
+  xparse-optarg-wrapped-value:
+    - clear_scopes: 1
+    - meta_scope: meta.function.parameters.default-value.latex meta.group.brace.latex
+    - include: brace-group-end
+    - include: general-constants
+    - include: general-commands
+    - include: macro-braces
```

Comment:
I've seen `#1` being supported in default values..

```suggestion
    - include: macro-braces
    - include: macro-placeholders
```

Reply:
Maybe more things like `tex-xonstants` are valid here?

Reply:
yes

Reply:
I'll change this to `macro-braces-content`, and add some tests